src/main/java/org/jvnet/hudson/test/InboundAgentRule.java

```diff
@@ -321,6 +324,7 @@ public void start(@NonNull JenkinsRule r, Options options) throws Exception {
         Objects.requireNonNull(name);
         stop(r, name);
         start(GetAgentArguments.get(r, name), options);
+        WaitForAgentOnline.wait(r, name);
```

Comment:
The Javadoc at the top of this file, which instructs consumers to call `r.waitOnline(agent)`, is now out of date, as are all cases of this pattern in Jenkins core.